lua/autorun/server/sv_pvp_movespeed.lua

```diff
@@ -36,19 +50,42 @@ local function movementMultiplier( totalWeight )
     return math.Clamp( multiplier, 0, 1 )
 end
 
+local function getBaseRunSpeed( ply )
+    return ply.CFC_PlyMS_BaseRunSpeed or normalRunSpeed
+end
+pvpMoveSpeed.getBaseRunSpeed = getBaseRunSpeed
+
+local function getBaseWalkSpeed( ply )
+    return ply.CFC_PlyMS_BaseWalkSpeed or normalWalkSpeed
+end
+pvpMoveSpeed.getBaseWalkSpeed = getBaseWalkSpeed
+
+local function alertAboutWeightSlowness( ply )
+    local now = RealTime()
+    local canAlertTime = ply.CFC_PlyMS_SlownessAlertReadyTime or 0
+
+    if now < canAlertTime then return end
+
+    ply:ChatPrint( "You are holding too many weapons! /drop some to regain speed." )
+    ply.CFC_PlyMS_SlownessAlertReadyTime = now + walkSpeedAlertCooldown
+end
+
 local function setSpeedFromWeight( ply, totalWeight )
     local multiplier = movementMultiplier( totalWeight )
+    local baseRunSpeed = getBaseRunSpeed( ply )
+    local baseWalkSpeed = getBaseWalkSpeed( ply )
 
     local newRunSpeed = baseRunSpeed * multiplier
     local newWalkSpeed = baseWalkSpeed * multiplier
-    ply:SetRunSpeed( math.max( newRunSpeed, minRunSpeed ) )
-    ply:SetWalkSpeed( math.max( newWalkSpeed, minWalkSpeed ) )
-
-    if newWalkSpeed < 100 then
-        ply:ChatPrint( "You are holding too many weapons! /drop some to regain speed." )
-        ply:SetCanWalk( false )
-    else
-        ply:SetCanWalk( true )
+    plyWraps.SetRunSpeed( ply, math.max( newRunSpeed, minRunSpeed ) )
+    plyWraps.SetWalkSpeed( ply, math.max( newWalkSpeed, minWalkSpeed ) )
+
+    local verySlow = newWalkSpeed < walkSpeedAlert
```

Comment:
walkSpeedAlert is necesarilly going to be slower than slow walking speed

Reply:
This is checking if the newly-calculated speed is slower than the alert threshold